etc/adminhtml/system.xml

```diff
@@ -0,0 +1,137 @@
+                    <comment>
+                        <![CDATA[Possibly used as an alternative to useStartTls (see above).]]>
+                    </comment>
+                    <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                </field>
+                <field id="bind_requires_dn" translate="label comment" type="select" sortOrder="60" showInDefault="1"
+                       showInWebsite="0" showInStore="0" canRestore="0">
+                    <label>Bind requires dn</label>
+                    <comment>
+                        <![CDATA[Required and must be `yes`, as OpenLDAP requires that usernames be in DN form when performing a bind.]]>
+                    </comment>
+                    <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                </field>
+                <field id="base_dn" translate="label comment" type="text" sortOrder="70" showInDefault="1"
+                       showInWebsite="0" showInStore="0" canRestore="0">
+                    <label>Base dn</label>
+                    <comment>
+                        <![CDATA[As with all servers, this option is required and indicates the DN under which all accounts being authenticated are located.]]>
+                    </comment>
+                </field>
+                <field id="bind_dn" translate="label comment" type="text" sortOrder="80" showInDefault="1"
+                       showInWebsite="0" showInStore="0" canRestore="0">
+                    <label>Bind dn</label>
+                    <comment>
+                        <![CDATA[Required and must be a DN, as OpenLDAP requires that usernames be in DN form when performing a bind. Try to use an unprivileged account.]]>
+                    </comment>
+                </field>
+                <field id="bind_password" translate="label comment" type="password" sortOrder="90" showInDefault="1"
+                       showInWebsite="0" showInStore="0" canRestore="0">
+                    <label>Bind password</label>
+                    <comment>
+                        <![CDATA[The password corresponding to the username above, but this may be omitted if the LDAP server permits an anonymous binding to query user accounts.]]>
+                    </comment>
+                </field>
+                <field id="allow_empty_password" translate="label" type="select" sortOrder="100" showInDefault="1"
+                       showInWebsite="0" showInStore="0" canRestore="0">
+                    <label>Allow empty password</label>
+                    <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                </field>
+                <field id="role" translate="label" type="select" sortOrder="120" showInDefault="1" showInWebsite="0"
+                       showInStore="0" canRestore="1">
+                    <label>Default Role</label>
+                    <source_model>Magenerds\Ldap\Model\Config\Role</source_model>
```

Comment:
That class should be RoleConfiguration. Also consider using Magenerds\Ldap\Model\System\Config\RoleConfiguration (System added)